src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/Tree.java

```diff
@@ -711,7 +705,7 @@ private void addAndSearch(@NotNull final World world)
         }
         for (int locX = locXMin; locX <= locXMax; locX++)
         {
-            for (int locY = locYMin; locY <= MAX_TREE_SIZE; locY++)
+            for (int locY = locYMin; !world.isYOutOfBounds(locY); locY++)
```

Comment:
world.func_234938_ad_() is the right call for this, yours is just the static version